PS - group all published instances

kalisp opened the pull request:
Closes: pypeclub/client#56

2.x version: https://github.com/pypeclub/OpenPype/pull/1415

kalisp commented:
For Pype 3 profiles in Settings will be used to allow this functionality for all hosts.

kalisp commented:
Implemented profiles, could be configured in Project Settings > Global > Publish > IntegrateAssetNew
![Screenshot 2021-04-29 175603](https://user-images.githubusercontent.com/4457962/116581146-51198880-a914-11eb-9814-6e9b3526edb7.png)

Currently implemented only `task` and `family` placeholders (and their uppercase variants). Any ideas about additional placeholders welcomed.

64qam commented:
Looks good

mkolar commented on openpype/plugins/publish/integrate_new.py:
```diff
@@ -716,6 +710,62 @@ def get_subset(self, asset, instance):
+        fill_pairs = None
+        if matching_profile:
+            template = matching_profile["template"]
+            fill_pairs = {
```
we should add more options here:

- subset
- host
- renderlayer (if present in the instance)

mkolar commented on openpype/plugins/publish/integrate_new.py:
```diff
@@ -716,6 +710,62 @@ def get_subset(self, asset, instance):
+                "family": filtering_criteria["families"],
+                "task": filtering_criteria["tasks"],
+                "Family": filtering_criteria["families"].capitalize(),
+                "Task": filtering_criteria["tasks"].capitalize()
```
maybe we could do this a bit less explicit, so it would look at the input and match the capitalisation of the value? I believe @iLLiCiTiT was already doing this same for creator somewhere.

kalisp commented:
Implemented new function `plugin_tools.prepare_template_data` to standardize preparation of data for template filling.

Implemented host, subset, renderlayer keys
